go-controller/pkg/factory/factory.go

```diff
@@ -1559,3 +1575,16 @@ type waitForCacheSyncer interface {
 func waitForCacheSyncWithTimeout(factory waitForCacheSyncer, stopCh <-chan struct{}) map[reflect.Type]bool {
 	return factory.WaitForCacheSync(util.GetChildStopChanWithTimeout(stopCh, types.InformerSyncTimeout))
 }
+
+// getEndpointSliceSelector returns an EndpointSlice selector function used in watchers.
+// When network segmentation is enabled it returns a selector that ignores EndpointSlices for headless services.
```

Comment:
@kyrtapz would you kindly explain why we ignore headless services when network segmentation is enabled ?

Reply:
We always ignore headless services, this is not new with this controller.
We ignore these because there is nothing to do there, they do not have a clusterIP and there is no loadbalancign to be performed:
https://kubernetes.io/docs/concepts/services-networking/service/#headless-services

Reply:
Right, I spaced out and forgot that a headless service does *not* have cluster IP. 

Tks.